contributors/design-proposals/core-metrics-pipeline.md

```diff
@@ -0,0 +1,320 @@
+
+Currently, cAdvisor collects a large number of metrics related to system and container performance. However, only some of these metrics are consumed by the kubelet summary API, and many are not used.  The kubelet [Summary API](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/api/v1alpha1/stats/types.go) is published to the kubelet summary API endpoint (stats/summary).  Some of the metrics provided by the summary API are consumed internally, but most are not used internally.
+
+### Motivations
+
+The [Monitoring Architecture](https://github.com/kubernetes/kubernetes/blob/master/docs/design/monitoring_architecture.md) proposal explains why a separate monitoring pipeline is required.
+
+cAdvisor is structured to collect metrics on an interval, which is appropriate for a stand-alone metrics collector.  However, many functions in the kubelet are latency-sensitive (eviction, for example), and would benifit from a more "On-Demand" metrics collection design.
+
+### Proposal
+
+I propose to use this set of core metrics, collected by the kubelet, and used solely by internal kubernetes components.
+
+### Non Goals
+
+Everything covered in the [Monitoring Architecture](https://github.com/kubernetes/kubernetes/blob/master/docs/design/monitoring_architecture.md) design doc.  This includes the third party metrics pipeline, and the methods by which the metrics found in this proposal are provided to other kubernetes components.
+
+Integration with CRI will not be covered in this proposal.  In future proposals, integrating with CRI may provide a better abstraction of information required by the core metrics pipeline to collect metrics.
+
+## Design
+
+This design covers only the internal Core Metrics Pipeline.
+
+High level requirements for the design are as follows:
+ - Do not break existing users.  We should continue to provide the full summary API as an optional add-on.  Once the monitoring pipeline is completed, the summary API will be provided by the monitoring pipeline, possibly through a stand-alone version of cAdvisor.
+ - The kubelet collects the minimum possible number of metrics for complete portable kubernetes functionalities.
+ - Metrics can be fetched "On Demand", giving the kubelet more up-to-date stats.
+
+Metrics requirements, based on kubernetes component needs, are as follows:
```

Comment:
I like the table you listed here for today's components. But there are more feature under-development / discussion, for example, initial resource, vertical pod autoscaling, resource estimation, etc. 

Instead of having which component use which metrics API, should we define which feature(s) use which metrics API. For example, resource-feasibility-checking and eviction-management are using the metrics API defined here, and other features use monitoring API (i.k.a. Summary API)?